test_interpreter.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import with_statement
+
+from interpreter import Interpreter
+from state import State
+from type_wrapper import Integer, List, String, NumericType
+from llvm_wrapper import *
+
+import sys
+import subprocess
+
+class InterpreterPlaceholder(Interpreter):
+
+    def __init__(self):
+        pass
+
+    def puts(self, string, args=[]):
+        self.output.append(string)
+        for arg in args:
+            if isinstance(arg, NumericType):
+                self.output.append(str(arg.value))
+            else:
+                self.output.append(str(arg))
+
+    def set_args(self, main_fun, global_state, argc, argv):
+        for index in range(0, LLVMCountParams(main_fun)):
+            param = LLVMGetParam(main_fun, index)
+            if index == 0:
+                global_state.set_variable(rffi.cast(rffi.INT, param),\
+                                                    Integer(argc))
+            else:
+                global_state.set_variable(rffi.cast(rffi.INT, param),\
+                                                    List(argv))
+
+    def setup_interp(self, filename, argc, argv):
+        self.output = []
+        assert len(self.output) == 0
+        module = LLVMModuleCreateWithName("test_module")
+
+        with lltype.scoped_alloc(rffi.CCHARPP.TO, 1) as out_message:
```

Comment:
I could only figure out how to assemble bytecode with llvm-as, which creates a file.